packages/storage-plugin/src/storage.module.ts

```diff
@@ -1,19 +1,40 @@
 import { NgModule, ModuleWithProviders } from '@angular/core';
-
 import { NGXS_PLUGINS } from '@ngxs/store';
+
 import { NgxsStoragePlugin } from './storage.plugin';
-import { NgxsStoragePluginOptions, NGXS_STORAGE_PLUGIN_OPTIONS } from './symbols';
+import {
+  NgxsStoragePluginOptions,
+  NGXS_STORAGE_PLUGIN_OPTIONS,
+  StorageOption,
+  STORAGE_ENGINE,
+  StorageEngine
+} from './symbols';
 
 export const defaultStoragePluginOptions: NgxsStoragePluginOptions = {
   key: '@@STATE',
-  storage: localStorage,
+  storage: StorageOption.LocalStorage,
   serialize: JSON.stringify,
   deserialize: JSON.parse
 };
 
+export function engineFactory(engine: StorageOption): StorageEngine {
+  if (engine === StorageOption.LocalStorage) {
+    return localStorage;
```

Comment:
We are still bound to the global context.  

```
return localStorage 
```

is the same as 
```
return global.localStorage
```

Where is the handling of the custom storage engine, let's say on mobiles via NativeScript.

Reply:
See StorageEngine